wagtail/wagtailcore/query.py

```diff
@@ -384,7 +384,9 @@ def specific_iterator(qs):
 
     # Yield all of the pages, in the order they occurred in the original query.
     for pk, content_type in pks_and_types:
-        yield pages_by_type[content_type][pk]
+        s = pages_by_type[content_type].get(pk)
+        if s is not None:
+            yield s
```

Comment:
In terms of performance, a `try…except` would be faster (though taking one more line).

Reply:
@BertrandBordage that's true, `try…except` would be faster, at least in case there's no `KeyError` otherwise… slower, but most of the time there will be no `KeyError` so it looks a good idea.